sycl/test/separate-compile/cpp_version_mismatch_test_2.cpp

```diff
@@ -0,0 +1,38 @@
+// >> ---- device compilation
+// RUN: %clangxx -std=c++14 -fsycl-device-only -Xclang -fsycl-int-header=sycl_ihdr_a.h %s -c -I %sycl_include
+
+// >> ---- code generation checks
+// RUN: FileCheck -input-file=sycl_ihdr_a.h %s
+// CHECK: #define STD_CPP_VERSION
+// CHECK-NEXT: #if (__cplusplus <= 201112L) && (STD_CPP_VERSION >= 201401L)
+// CHECK-NEXT: #error "C++ version (std=c++11 or less) for host compilation
```

Comment:
The wording needs work, but this is perhaps less harmful if it is a pragma-warning (assuming this isn't included as a system header).